### PR TITLE
RandomPlayAnimation

### PR DESCRIPTION
**Pull request by GraphicEdit**
Randomizing Animation...
"play-show" randomly "small" part(s) on background.

### CODE REVIEW

File: src/ProjectNeon/Assets/VFX/RandomPlayAnimation.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+﻿using UnityEngine;
+using System.Collections.Generic;
+using System.Collections;
+
+public class RandomPlayAnimation : MonoBehaviour {
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/VFX/RandomPlayAnimation.cs]: Braces belong at the start of a line in C#.

File: src/ProjectNeon/Assets/VFX/RandomPlayAnimation.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ 
+	public void Start()
+    { 
+		this.gameObject.GetComponent<SpriteRenderer> ().enabled = false;
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/VFX/RandomPlayAnimation.cs]: `GetComponent` is an expensive call. Should only happen once on Awake.

File: src/ProjectNeon/Assets/VFX/RandomPlayAnimation.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+		StartCoroutine(Wait());
+	}     
+
+	IEnumerator  Wait()
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/VFX/RandomPlayAnimation.cs]: Should put an explicit `private` modifier on this method.

File: src/ProjectNeon/Assets/VFX/RandomPlayAnimation.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    {
+		yield return new WaitForSeconds(Random.Range(5, 20)); //for waiting
+		//InvokeRepeating("OnOf", Random.Range(10, 20), Random.Range(10, 20)); //for showing (time-repeat)
+		OnOf();
+	}
+	
+	public void OnOf()
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/VFX/RandomPlayAnimation.cs]: Usually we call this `Toggle`. If you want to use the words "On" and "Off" then it should be the whole word "Off"

**Comment by GraphicEdit** [comment]
all done..

File: src/ProjectNeon/Assets/VFX/RandomPlayAnimation.cs [hunk 5]
```diff
@@ -0,0 +1,33 @@
+﻿using UnityEngine;
+using System.Collections.Generic;
+using System.Collections;
+
+public class RandomPlayAnimation : MonoBehaviour 
+{
+ 
+	public void Awake()
+    { 
+		this.gameObject.GetComponent<SpriteRenderer> ().enabled = false;
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/VFX/RandomPlayAnimation.cs]: There are still four calls to` GetComponent` in this class.

What you want to do is cache the component.

```

private SpriteRenderer _renderer;

public void Awake() 
{
    _renderer = GetComponent<SpriteRenderer>();
}
```

And then everytime you would get the component, you just use `_renderer.enabled`... etc.

Also, anytime you have a Script that requires another component in this way, it's best to put a RequiresComponent attribute on the Script like this: https://docs.unity3d.com/ScriptReference/RequireComponent.html

**Comment by GraphicEdit** [comment]
Ok... Can you look at it please.. thanks

**Pull request merged**